### ISSUE TITLE
Feature Request: On-Click  Command Support in Hyprlock Widget

### ISSUE THREAD

**Issue by ashish-kus** [hyprwm/hyprlock#366]
I hope this message finds you well. I am writing to request a new feature for Hyprlock that would significantly enhance its functionality and user experience.

## Feature Request:
I would like to propose the addition of an on-click bash command feature within the widget in Hyprlock. This functionality would allow users to execute bash commands directly from the lockscreen, enabling actions such as **shutdown, reboot, logout, and sleep** with a single click.

#### Suggested Implementation:
- Add an option in the widget configuration to specify bash commands for different on-click actions.
- Ensure secure execution of these commands to prevent any potential misuse
    
    
Example configuration snippet:
``` 
widget_name {
monitor =
# further options
on-click = systemctl syspend 
}
```

**Comment by Eclextic** [comment]
Well, what is secure execution? I’m totally on board, but what do you propose as a security measure?

**Comment by Noino** [comment]
Isn't a button on the lock screen by definition unauthorized, so whatever the user here exposes is on them imho. 

Totally want a suspend button though that i can press even though I'm carrying a child or two

**Comment by Eclextic** [comment]
I think maybe it's better to have a list of commands that can be executed and only those?

Mainly for security as well as testability with bugs, minimizing them etc...

And we could have it so that there is another variable that allows bash script execution, but whenever that is used, we get a warning and need to supply a flag or smth...


So what I propose:
2 Variables. 1 uses a list of known simple bash commands that are known to be safe and can't be executed without notifying the user.
2 allows for bash script execution, but using them forces a flag to make sure that the user is responsible for what is about to happen next. (Then again bash scripts already get executed with no verification sooo... Make of it what you will.)

**Comment by bvr-yr** [comment]
security what? it's totally responsibility of users and their own config, if they wish it could be even `rm -rf --no-preserve-root /`, which you even now technically can execute in label `cmd` for example. By simply clicking button, you can't do anything beyond what is already defined and therefor permitted by the user in config

**Comment by SERDUSSS** [comment]
This feature would open an infinite number of possibilities and I think these should be kept without any limitation

Also this should not suppose a unauthorized command execution since as @Noino said, if you let someone execute a command without being authenticated you should be sure that command will not cause a threat to the system and that it cannot be abused in any way

**Comment by bvr-yr** [comment]
please explain me how can one abuse anthing that is permitted by user since this command is predefined in config, I can't get it

**Comment by ejiektpobehuk** [comment]
Media controls are a possible use case. Mobile phones provide them on a locked screen. I'm not sure about the threat model there

**Comment by Eclextic** [comment]
No, this guy means that `sudo` commands won't work (unless you start hyprlock with sudo), which is honestly a very much a valid argument.

Also who uses hyprlock on a phone? xd

**Comment by ejiektpobehuk** [comment]
Got it, thanks!

It's possible to configure polkit to enable poweroff, reboot and suspend for a regular user without any authentication. This way hyprlock can run whatever with user permissions and it's users job to bring power related functionality to the user level.

ps. I do have a PinePhone Pro but I haven't tried to run hyprland there. Might be a cool experiment =]

**Comment by KP2048** [comment]
Is this still being considered? Would be a very useful thing

**Comment by LorenzHohermuth** [comment]
I would like this feature, can you please merge

thank you

**Comment by Eclextic** [comment]
tis a request not a feature. merge impossible comrade...

**Comment by Noino** [comment]
so whats happening? we even have some code but everything seems stalled. 
What to do?

**Comment by PaideiaDilemma** [comment]
We got a contribution that only needs some cleanup. I can take over the MR if it isn't done by the author, so don't worry.

Give the people some time!

**Comment by PaideiaDilemma** [comment]
any testing of https://github.com/hyprwm/hyprlock/issues/695 would be appreciated!

**Comment by ashish-kus** [comment]
How do we use this feature ? I could not understand through WIKI. @PaideiaDilemma
